### PR TITLE
Initial sample app

### PR DESCRIPTION
**Pull request by mcasper3**
Let me know if you think I should add a more complex example that uses a custom implementation of AdapterPresenter or some other features

### CODE REVIEW

File: konveyor-sample/build.gradle [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    buildToolsVersion "25.0.2"
+
+    defaultConfig {
+        applicationId "com.example.konveyor"
+        minSdkVersion 15
+        targetSdkVersion 25
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
```
Comment (philipuvarov) [inline konveyor-sample/build.gradle]: Do we need this?

File: konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.example.konveyor
+
+import com.avito.konveyor.blueprint.Item
+import com.avito.konveyor.data_source.DataSource
+import com.example.konveyor.items.article.Article
+import com.example.konveyor.items.photo.Photo
+
+class LocalDataSource : DataSource<Item> {
+    private val items = listOf(
+            Article(1, "Northern Lights", R.string.article_body_preview, R.drawable.article1),
```
Comment (philipuvarov) [inline konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt]: Shall we use named parameter here for id?

File: konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.example.konveyor
+
+import com.avito.konveyor.blueprint.Item
+import com.avito.konveyor.data_source.DataSource
+import com.example.konveyor.items.article.Article
+import com.example.konveyor.items.photo.Photo
+
+class LocalDataSource : DataSource<Item> {
```
Comment (philipuvarov) [inline konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt]: Why don't you use ListDataSource, that is provided by the library?

File: konveyor-sample/src/main/java/com/example/konveyor/items/article/Article.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.example.konveyor.items.article
+
+import android.support.annotation.DrawableRes
+import android.support.annotation.StringRes
+import com.avito.konveyor.blueprint.Item
+
+data class Article(override val id: Long, val title: String, @StringRes val bodyPreview: Int, @DrawableRes val image: Int) : Item
```
Comment (philipuvarov) [inline konveyor-sample/src/main/java/com/example/konveyor/items/article/Article.kt]: Why is it a data class? We don't use copy constructors or even equals/hashcode. Looks like unnecessary overhead.

File: konveyor-sample/src/main/java/com/example/konveyor/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        itemList.layoutManager = LinearLayoutManager(this)
+        itemList.addItemDecoration(VerticalSpaceItemDecoration(itemOffset))
+
+        val binder = ItemBinder.Builder()
+                .registerItem(ArticleBlueprint(ArticlePresenter()))
+                .registerItem(PhotoBlueprint(PhotoPresenter()))
+                .build()
+
+        val presenter = SimpleAdapterPresenter(binder, binder)
+        itemList.adapter = SimpleRecyclerAdapter(presenter, binder)
```
Comment (philipuvarov) [inline konveyor-sample/src/main/java/com/example/konveyor/MainActivity.kt]: hasStableIds = true?

**Review by philipuvarov — CHANGES_REQUESTED** [review]
Great work on the sample app! Left a couple of comments. 

As for the example itself, I guess it perfectly serves its purpose, that is provides an example of the most basic implementation. Though, other, more complex, examples would be useful as well. Just create a separate issue for it, so we don't do double work :)

File: konveyor-sample/src/main/res/layout/view_photo.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            android:scaleType="fitStart"
+            />
+
+        <TextView
+            android:id="@+id/photoTitle"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="4dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginRight="8dp"
```
Comment (philipuvarov) [inline konveyor-sample/src/main/res/layout/view_photo.xml]: Shall we extract this dimension? Looks like it is repeating here and there.

**Comment by philipuvarov** [comment]
@mcasper3 by the way, the images. Are they free to use?

File: konveyor-sample/build.gradle (hunk 1 above)
Comment (mcasper3) [reply konveyor-sample/build.gradle]: No, I forgot to remove that. Thanks for pointing it out

File: konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt (hunk 3 above)
Comment (mcasper3) [reply konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt]: Oops, I'll switch it over to using that

File: konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt (hunk 2 above)
Comment (mcasper3) [reply konveyor-sample/src/main/java/com/example/konveyor/LocalDataSource.kt]: We could; we'd also have to convert the other parameters to being named as well, though.

**Comment by mcasper3** [comment]
@Kistamushken yes, the photos are from https://unsplash.com/

**Pull request merged**